mantidimaging/main.py

```diff
@@ -93,42 +94,66 @@ def setup_application() -> QApplication:
 
 def main() -> None:
     args = parse_args()
-    # Print version number and exit
+
     if args.version:
         from mantidimaging import __version__ as version_no
-
         print(version_no)
         return
 
     q_application = setup_application()
-
+    show_unclean_shutdown_warning(settings)
     path = args.path if args.path else ""
     operation = args.operation if args.operation else ""
 
-    CommandLineArguments(path=path,
-                         operation=operation,
-                         show_recon=args.recon,
-                         show_live_viewer=args.live_viewer,
-                         show_spectrum_viewer=args.spectrum_viewer)
+    CommandLineArguments(
+        path=path,
+        operation=operation,
+        show_recon=args.recon,
+        show_live_viewer=args.live_viewer,
+        show_spectrum_viewer=args.spectrum_viewer,
+    )
 
     h.initialise_logging(args.log_level)
-
-    settings = QSettings()
     process_count = settings.value("multiprocessing/process_count", 8, type=int)
+    settings.setValue("app/last_shutdown_clean", False)
 
     from mantidimaging import gui
     try:
         pm.create_and_start_pool(process_count)
         gui.execute()
         result = q_application.exec_()
-    except BaseException as e:
-        if sys.platform == 'linux':
-            pm.clear_memory_from_current_process_linux()
-        raise e
```

Comment:
The `except` is still needed to call the `clear_memory_from_current_process_linux()` function